pkg/R/DataFrame.R

```diff
@@ -774,21 +774,85 @@ setMethod("foreachPartition",
           })
 
 
-############################## DSL ##################################
+############################## SELECT ##################################
+
+getColumn <- function(x, c) {
+  column(callJMethod(x@sdf, "col", c))
+}
 
 setMethod("$", signature(x = "DataFrame"),
           function(x, name) {
-            column(callJMethod(x@sdf, "col", name))
+            getColumn(x, name)
+          })
+
+setMethod("$<-", signature(x = "DataFrame"),
+          function(x, name, value) {
+            stopifnot(class(value) == "Column")
```

Comment:
One risk with supporting non-column types is that users might expect behavior similar to R data frames. In R if you do 
```
df$age <- "5"
```
It sets the value "5" to all the rows. Since we don't want to support this (changing all the rows might be very expensive ?) I am fine with keeping this only for columns.